front/app/services/navbar.ts

```diff
@@ -81,9 +72,24 @@ export async function toggleEvents({ enabled }: { enabled: boolean }) {
   return response;
 }
 
-export function navbarItemIsEnabled(
-  navbarItems: INavbarItem[],
-  code: TDefaultNavbarItemCode
+// utility function to get slug associated with navbar item
+export function getNavbarItemSlug(
+  navbarItemCode: TNavbarItemCode,
+  pageBySlugId: TPageSlugById,
+  pageId?: string
 ) {
-  return navbarItems.some((navbarItem) => code === navbarItem.attributes.code);
+  // Default navbar item
+  if (navbarItemCode !== 'custom' && !pageId) {
+    return DEFAULT_PAGE_SLUGS[navbarItemCode];
+  }
+
+  // Page navbar item
+  if (navbarItemCode === 'custom' && pageId) {
+    return pageBySlugId[pageId];
+  }
+
+  // This is impossible, but I can't seem to make typescript understand
+  // that. So just returning an empty string here so that the function
+  // return type is typed correctly
+  return '';
```

Comment:
This code is unreachable, since the two if statements already cover all possible cases. This return statement can only be reached if there is invalid data in the database. The question is then what we want as fallback behavior when the database is invalid. An empty string means that we will redirect to the home page. I also considered just throwing an error here... what do you think is best?

Reply:
I think having the null value will force you to do checks in the place where you use getNavbarItemSlug's return value. So also for people using it in the future, it'll be less likely to introduce bugs. And it's semantically more correct. Our slug isn't '' but null (or undefined) if none of the conditions match. 

And you can still redirect to the home page if you have null as the fallback value. :-)

Reply:
Sure, but then what do we do when it's null?

Reply:
To be decided for each case?

Reply:
Alright, I will return `null` here. In the places where we call this (desktop navbar and mobile navbar for slug generation, Navigation settings page) I will just make it direct to home for now